src/queryMetrics/queryMetrics.ts

```diff
@@ -0,0 +1,369 @@
+        const queryPreparationTimesArray = [];
+        let indexLookupTime = TimeSpan.zero;
+        let documentLoadTime = TimeSpan.zero;
+        let vmExecutionTime = TimeSpan.zero;
+        const runtimeExecutionTimesArray = [];
+        let documentWriteTime = TimeSpan.zero;
+        const clientSideQueryMetricsArray = [];
+
+        queryMetricsArray.push(this);
+
+        for (const queryMetrics of queryMetricsArray) {
+
+            if (queryMetrics == null) {
+                throw new Error("queryMetricsArray has null or undefined item(s)");
+            }
+
+            retrievedDocumentCount += queryMetrics.retrievedDocumentCount;
+            retrievedDocumentSize += queryMetrics.retrievedDocumentSize;
+            outputDocumentCount += queryMetrics.outputDocumentCount;
+            outputDocumentSize += queryMetrics.outputDocumentSize;
+            indexHitDocumentCount += queryMetrics.indexHitDocumentCount;
+            totalQueryExecutionTime = totalQueryExecutionTime.add(queryMetrics.totalQueryExecutionTime);
+            queryPreparationTimesArray.push(queryMetrics.queryPreparationTimes);
+            indexLookupTime = indexLookupTime.add(queryMetrics.indexLookupTime);
+            documentLoadTime = documentLoadTime.add(queryMetrics.documentLoadTime);
+            vmExecutionTime = vmExecutionTime.add(queryMetrics.vmExecutionTime);
+            runtimeExecutionTimesArray.push(queryMetrics.runtimeExecutionTimes);
+            documentWriteTime = documentWriteTime.add(queryMetrics.documentWriteTime);
+            clientSideQueryMetricsArray.push(queryMetrics.clientSideMetrics);
+        }
+
+        return new QueryMetrics(
+            retrievedDocumentCount,
+            retrievedDocumentSize,
+            outputDocumentCount,
+            outputDocumentSize,
+            indexHitDocumentCount,
+            totalQueryExecutionTime,
+            QueryPreparationTimes.createFromArray(queryPreparationTimesArray),
+            indexLookupTime,
+            documentLoadTime,
+            vmExecutionTime,
+            RuntimeExecutionTimes.createFromArray(runtimeExecutionTimesArray),
+            documentWriteTime,
+            ClientSideMetrics.createFromArray(clientSideQueryMetricsArray));
+    }
+
+    /**
+     * Output the QueryMetrics as a delimited string.
+     * @memberof QueryMetrics
+     * @instance
+     * @ignore
+     */
+    public toDelimitedString() {
+        return QueryMetricsConstants.RetrievedDocumentCount + "=" + this.retrievedDocumentCount + ";"
+            + QueryMetricsConstants.RetrievedDocumentSize + "=" + this.retrievedDocumentSize + ";"
+            + QueryMetricsConstants.OutputDocumentCount + "=" + this.outputDocumentCount + ";"
+            + QueryMetricsConstants.OutputDocumentSize + "=" + this.outputDocumentSize + ";"
+            + QueryMetricsConstants.IndexHitRatio + "=" + this.getIndexHitRatio() + ";"
+            + QueryMetricsConstants.TotalQueryExecutionTimeInMs +
```

Comment:
If possible can we keep this in one line, since the format is key = value. If the line is too long, then we can leave it as is.